### PR TITLE
fix: build-script for ubuntu 16.04 and fedora 30

### PR DESCRIPTION
**Pull request by amitsadaphule**
#### Description of Change

Fixes the build for Fedora 30 and Ubuntu 16.04 (https://github.com/leo-lb/electron/issues/1).

#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

#### Release Notes

Notes: `no-notes`

### CODE REVIEW

File: ppc64le_build.bash [hunk 1]
```diff
@@ -1,22 +1,50 @@
...
 
 set -eux
 
-#apt-get update
-#apt-get install -y build-essential clang git vim cmake python libcups2-dev pkg-config libnss3-dev libssl-dev libglib2.0-dev libgnome-keyring-dev libpango1.0-dev libdbus-1-dev libatk1.0-dev libatk-bridge2.0-dev libgtk-3-dev libkrb5-dev libpulse-dev libxss-dev re2c subversion curl libasound2-dev libpci-dev mesa-common-dev gperf bison uuid-dev clang-format libatspi2.0-dev libnotify-dev libgconf2-dev libcap-dev libxtst-dev libxss1 python-dbusmock openjdk-8-jre ninja-build clang-format wget
+. /etc/os-release
+OS=$NAME
+OSVER=$VERSION_ID
+
+if [[ $OS == "Ubuntu" ]] && [[ $OSVER == "16.04" ]]
```
Comment (llebout) [inline ppc64le_build.bash]: While it's good you choose to add support for Ubuntu Xenial, can't we keep Ubuntu Bionic even if it couldnt build Chromium lately?

File: ppc64le_build.bash [hunk 2]
```diff
@@ -1,22 +1,50 @@
...
+    exit 0
+fi
 
 VERSION=v10.20.1
 DISTRO=linux-ppc64le
 
 wget "https://nodejs.org/dist/$VERSION/node-$VERSION-$DISTRO.tar.xz"
 tar -xJvf node-$VERSION-$DISTRO.tar.xz
 PATH=$(pwd)/node-$VERSION-$DISTRO/bin:$PATH
+ln -sf $(pwd)/node-$VERSION-$DISTRO/bin/node /usr/bin/node
```
Comment (llebout) [inline ppc64le_build.bash]: You should not create a link in a system directory, manipulating the PATH variable and patching build scripts if they reference absolute paths should be sufficient.

We do not want the build script to require being run as root

**Review by llebout — CHANGES_REQUESTED** [review]
Few comments, after that it's all good!

File: ppc64le_build.bash (hunk 1 above)
Comment (amitsadaphule) [reply ppc64le_build.bash]: Thanks for the review @leo-lb! The reason why I didn't add bionic was that it had build issues and I couldn't really get it to build/test. But as you said, I'll just let bionic build too to enter there.

File: ppc64le_build.bash (hunk 2 above)
Comment (amitsadaphule) [reply ppc64le_build.bash]: OK, I'll try to fix the reference to absolute path '/usr/bin/node' in the build. Same thing will apply for the python3.7 ubuntu link too. Will work on that one too.

Comment (llebout) [reply ppc64le_build.bash]: Didnt see the Python one, yes please do!

File: ppc64le_build.bash (hunk 1 above)
Comment (llebout) [reply ppc64le_build.bash]: "enter here" you mean in the Xenial code path? Dependencies may be different. - Just add a condition that retains the old code that installed dependencies, is there any problem with that?

Comment (amitsadaphule) [reply ppc64le_build.bash]: OK, it seems that I had misinterpreted your comment. I'll add separate condition for bionic. Also, the issue is not with the dependencies, I've used mostly the same dependencies for xenial (except for python3.7 and ninja). @NishikantThorat had tried the bionic build and had faced issues in the build, so we didn't really pursue it further.

Comment (amitsadaphule) [reply ppc64le_build.bash]: I was able to build on bionic too.

File: ppc64le_build.bash (hunk 2 above)
Comment (amitsadaphule) [reply ppc64le_build.bash]: Completed the necessary changes. Awaiting approval for contributing to https://github.com/leo-lb/chromium_power (specifically to https://github.com/leo-lb/chromium_power/blob/master/third_party/0001-DONTMERGE-third_party-node-Use-system-nodejs-binary.patch) w.r.t node path.

Comment (amitsadaphule) [reply ppc64le_build.bash]: Please review https://github.com/leo-lb/chromium_power/pull/3 for the node path fix.

**Comment by amitsadaphule** [comment]
The PR https://github.com/leo-lb/chromium_power/pull/3 along with the new commit a66b40b3afcaff3f261300c66759e43f41216c7d incorporates all the review comments.

**Comment by llebout** [comment]
@amitsadaphule Merged! As commented on the PR;

You will also need to update the submodule in this electron fork, by doing "git pull" when in the chromium_power subdirectory, committing and pushing. 

It's the "patches" directory

**Pull request merged**